src/App.tsx

```diff
@@ -15,48 +16,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET,
+  LENGTH,
+  NONE,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (SortType.NONE !== sortType) {
+    visibleGoods.sort((word1, word2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return word1.localeCompare(word2);
+
+        case SortType.LENGTH:
+          return word1.length - word2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const reset = () => {
```

Comment:
Pay attention to the checklist
<img width="1125" alt="image" src="https://user-images.githubusercontent.com/91411692/235118135-ac379082-8ed0-4989-8ac6-344d9356df95.png">

```suggestion
  const [isReversed, setIsReversed] = useState(false);
  
  const reset = () => {
```